installer/functions.sh

```diff
@@ -439,8 +439,8 @@ iscontainer() {
 issystemd() {
     local pids p myns ns systemctl
 
-    # if the directory /etc/systemd/system does not exit, it is not systemd
-    [ ! -d /etc/systemd/system ] && return 1
+    # if the directory /lib/systemd/system does not exit, it is not systemd
+    [ ! -d /lib/systemd/system ] && return 1
```

Comment:
Many systemd systems, including ubuntu, run `systemd` as `init`, so `pidof systemd` does not work.

Then, `command -v systemctl` may not work, since many systems prefer to supply the `service` command that is a wrapper to an non path accessible `systemctl`. I believe I encountered this once, but I don't recall which distro it was.

Reply:
Now days `systemctl` is probably the standard. Over the last years, distros got creative with systemd though...

Reply:
Strange process list from ubuntu 18.04:

```
    1 ?        Ss     0:01 /sbin/init
  434 ?        S<s    0:00 /lib/systemd/systemd-journald
  442 ?        Ss     0:00 /sbin/lvmetad -f
  458 ?        Ss     0:00 /lib/systemd/systemd-udevd
  642 ?        Ssl    0:00 /lib/systemd/systemd-timesyncd
  656 ?        Ss     0:00 /lib/systemd/systemd-networkd
  686 ?        Ss     0:00 /lib/systemd/systemd-resolved
  813 ?        Ss     0:00 /usr/sbin/atd -f
  816 ?        Ss     0:01 /sbin/iscsid
  817 ?        S<Ls   0:00 /sbin/iscsid
  819 ?        Ssl    0:01 /usr/lib/accountsservice/accounts-daemon
  828 ?        Ss     0:00 /lib/systemd/systemd-logind
  832 ?        Ssl    0:00 /usr/bin/python3 /usr/bin/networkd-dispatcher --run-startup-triggers
  837 ?        Ss     0:00 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
  844 ?        Ssl    0:05 /usr/lib/snapd/snapd
  852 ?        Ssl    0:03 /usr/bin/lxcfs /var/lib/lxcfs/
  863 ?        Ss     0:00 /usr/sbin/cron -f
  866 ?        Ssl    0:00 /usr/sbin/rsyslogd -n
  868 ?        Ssl    0:02 /usr/sbin/irqbalance --foreground
  882 ?        Ssl    0:00 /usr/lib/policykit-1/polkitd --no-debug
  904 tty1     Ss+    0:00 /sbin/agetty -o -p -- \u --noclear tty1 linux
  911 ?        Ss     0:00 /usr/sbin/sshd -D
 1115 ?        Ss     0:00  \_ sshd: costa [priv]
 1245 ?        S      0:00      \_ sshd: costa@pts/0
 1246 pts/0    Ss     0:00          \_ -bash
 1262 pts/0    S      0:00              \_ sudo su
 1265 pts/0    S      0:00                  \_ su
 1266 pts/0    S      0:00                      \_ bash
 1278 pts/0    R+     0:00                          \_ ps fax
 7139 ?        Ssl   11:09 /usr/sbin/netdata -P /run/netdata/netdata.pid -D
29541 ?        S      0:26  \_ /usr/libexec/netdata/plugins.d/apps.plugin 1
  900 ?        S      0:00  \_ bash /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1
 1120 ?        Ss     0:00 /lib/systemd/systemd --user
 1121 ?        S      0:00  \_ (sd-pam)
```

and 

```sh
# ls -l /proc/1/exe 
lrwxrwxrwx 1 root root 0 Sep  9 22:16 /proc/1/exe -> /lib/systemd/systemd
```

So, what saves the day is this:

```
1120 ?        Ss     0:00 /lib/systemd/systemd --user
```

So, I think we should remove this check:

```sh
    # if systemd is not running, it is not systemd
    pids=$(pidof systemd 2>/dev/null)
    [ -z "${pids}" ] && return 1
```

Reply:
yes, this one is not needed